### DISCUSSION TITLE
Way to query what network WLAN config options are implemented?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#9110] [Using MicroPython] [answered]
Way to query what network WLAN config options are implemented?  Came across this just this morning, that config('reconnects') exists, but on the latest nightly build for ESP8266, I get ValueError: unknown config param.  But a Google search will bring up quite a few hits that such parameter exists, for ESP32 and ESP8266, or are all those hits bogus?

**Comment by robert-hh** [comment, marked as answer]
This is what the documentation tells: https://docs.micropython.org/en/latest/library/network.WLAN.html?highlight=wlan%20config#network.WLAN.ifconfig
A glance through the ESP8266 code reveals:
For setting
```
mac
ssid
essid
hidden
security
authmode
key
password
channel
hostname
dhcp_hostname
protocol
```
For requesting
```
mac
ssid
essid
hidden
security
authmode
channel
hostname
dhcp_hostname
protocol
```

**Reply by robert-hh** [reply]
The parameter reconnects is available at the ESP32 port.

**Reply by jimmo** [reply]
We're working on clarifying the port-specific exceptions in the docs... I chip away at this a few hours each week. This is a good example, similarly the fields in the tuple returned from scan, or auth mode, etc.

It's a difficult balance, some chips just have more features, hence why we expose additional params. But absolutely the docs should clarify this.

**Comment by Jibun-no-Kage** [comment]
Any and all improvements, greatly appreciated.  :)

**Comment by scruss** [comment]
I run this when I'm not sure:
```
# check WLAN config values

import network
sta_if = network.WLAN(network.STA_IF)

for i in "authmode channel dhcp_hostname essid hidden hostname mac protocol security ssid".split():
    try:
        cfg=sta_if.config(i)
    except:
        cfg="unknown config parameter"
    print(i, ":", cfg)
```
This assumes that network's already set up in boot.py, or at least in some code that's already been run.

**Comment by Jibun-no-Kage** [comment]
Yeah, that is a cool method, I did some similar, creating a dictionary object that I can then query.
